docs/CORE_DEV_GUIDE.md

```diff
@@ -0,0 +1,180 @@
+experience. Repeat yourself liberally, and, if you don’t recognize a
+contributor, point them to our development guide, or other GitHub
+workflow tutorials around the web. Do not assume that they know how
+GitHub works (e.g., many don't realize that adding a commit
+automatically updates a pull request). Gentle, polite, kind
+encouragement can make the difference between a new core developer and
+an abandoned pull request.
+
+When reviewing, focus on the following:
+
+1. **Usability and generality:** `napari` is an GUI application that strives to be accessible
+to both coding and non-coding users, and new features should ultimately be
+accessible to everyone using the app. `napari` targets the scientific user
+community broadly, and core features should be domain agnostic and general purpose.
+Custom functionality is meant to be provided through our plugin ecosystem. If in doubt
+consult back with our [mission and values](MISSION_AND_VALUES.md).
+
+2. **Performance and benchmarks:** As `napari` targets scientific applications which often involve
+large multidimensional datasets, high performance is a key value of `napari`. While
+every new feature won't scale equally to all sizes of data keeping in mind performance
+and our [benchmarks](BENCHMARKS.md) during a review may be important, and you may
+need to ask for benchmarks to be run and reported or new benchmarks to be added.
+
+3. **APIs and stability:** Coding users and plugin developers will make
+extensive use of our APIs. The foundation of a healthy plugin ecosystem will be
+a fully capable and stable set of APIs and so as the `napari` matures it will
+very important to ensure our APIs are stable. For now, while the project is still
+in an earlier stage spending the extra time to consider names of public facing
+variables and methods, along side function signatures, could save us considerable
+trouble in the future.
+
+4. **Documentation and tutorials:** All new methods should have appropriate doc
+strings following [PEP257](https://www.python.org/dev/peps/pep-0257/) and the
+[NumPy documentation guide](https://docs.scipy.org/doc/numpy/docs/howto_document.html).
+For any major new features accompanying changes should be made to our
+[tutorials repository](https://github.com/napari/napari-tutorials), that not only
+illustrates the new feature, but explains it.
+
+5. **Implementations and algorithms:** You should understand the code being modified
+or added before approving it.  (See `Merge Only Changes You Understand` below.)
+Implementations should do what they claim, and be simple, readable, and efficient.
+
+6. **Tests:** All contributions *must* be tested, and each added line of code
+should be covered by at least one test. Good tests not only execute the code,
+but explores corner cases.  It can be tempting not to review tests, but please
+do so.
+
+Other changes may be *nitpicky*: spelling mistakes, formatting,
+etc. Do not insist contributors make these changes, and instead you can
```

Comment:
oh no I feel called out!

Reply:
Overall I'm supportive, but should this apply to both core-dev originating PRs and user generated PRs too? Reviewing could become a bigger burden if it applies to core-dev PRs and could slow things down. For user generated PRs I've had trouble pushing commits directly to PRs.

Reply:
I'm still not sure about this one. I also worry about always doing these things for people rather than giving them the option first. @jni how about
```
but instead you should offer to make these changes by ... and be prepared to do them yourself if need.
```

Reply:
*if needed

But yes, that's good. =)

Reply:
ok have updated